packit/specfile.py

```diff
@@ -16,12 +18,14 @@
 
 
 class Specfile(SpecFile):
-    def __init__(self, path="", dir=None):
+    def __init__(self, path: Union[str, Path], sources_dir: Union[str, Path] = ""):
         s = inspect.signature(SpecFile)
         if "changelog_entry" in s.parameters:
-            super().__init__(path=path, sources_location=str(dir), changelog_entry="")
+            super().__init__(
+                path=str(path), sources_location=str(sources_dir), changelog_entry=""
+            )
         else:
-            super().__init__(path=path, sources_location=str(dir))
+            super().__init__(path=str(path), sources_location=str(sources_dir))
```

Comment:
:+1: 

It would be nice to use `Path` internally everywhere and need the retyping only for other libraries.

Reply:
#733

Reply:
Thanks.